### PR TITLE
[Tutorial/Basic] 123 hoc

### PR DESCRIPTION
**Pull request by qkreltms**
## Changes
Change text color by using color property which given from HOC.

### References
#123 

Reviewers: @sonypark, @Haerang94

### CODE REVIEW

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/pages/todoListPage/todoList/Item.js [hunk 1]
```diff
@@ -24,34 +55,4 @@ const Item = ({ todo, onRemove, onCheck }) => {
...
-        marginHorizontal: 10,
-    },
-    trash: {
-        marginVertical: 10,
-        marginHorizontal: 10,
-    },
-});
-
-export default Item;
+export default withRainbow(Item);
```
Comment (sonypark) [inline Examples/BasicTutorial/AdvancedFeatures/07-hoc/pages/todoListPage/todoList/Item.js:59]: I think you didn't import `withRainbow`, didn't you get any errors?

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,211 @@
+apply plugin: "com.android.application"
+
+import com.android.build.OutputFile
+
+/**
+ * The react.gradle file registers a task for each build variant (e.g. bundleDebugJsAndAssets
+ * and bundleReleaseJsAndAssets).
+ * These basically call `react-native bundle` with the correct arguments during the Android build
```
Comment (chococrunch2021) [inline Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle:8]: I guess this sentence would be 'These are basically called' or 'They basically call'

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+ *   nodeExecutableAndArgs: ["node"],
+ *
+ *   // supply additional arguments to the packager
+ *   extraPackagerArgs: []
+ * ]
+ */
+
+project.ext.react = [
+    entryFile: "index.js",
+    enableHermes: false,  // clean and rebuild if changing
```
Comment (chococrunch2021) [inline Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle:80]: I guess if it is changed or if it changes , if it is changes would be better

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+]
+
+apply from: "../../node_modules/react-native/react.gradle"
+apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"
+
+/**
+ * Set this to true to create two separate APKs instead of one:
+ *   - An APK that only works on ARM devices
+ *   - An APK that only works on x86 devices
+ * The advantage is the size of the APK is reduced by about 4MB.
```
Comment (chococrunch2021) [inline Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle:90]: The advantage is that the size of ~

**Review by chococrunch2021 — COMMENTED** [review]
I can't sure all of my reviews are correct, but just take this into considering.
Thanks.

**Review by chococrunch2021 — DISMISSED** [review]

**Review by JeffGuKang — CHANGES_REQUESTED** [review]
There is some requests by other reviewers.

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle (hunk 2 above)
Comment (qkreltms) [reply Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle:8]: This is auto generated comment by react native.

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle (hunk 3 above)
Comment (qkreltms) [reply Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle:80]: This is auto generated comment by react native.

File: Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle (hunk 4 above)
Comment (qkreltms) [reply Examples/BasicTutorial/AdvancedFeatures/07-hoc/android/app/build.gradle:90]: This is auto generated comment by react native.

**Review by JeffGuKang — APPROVED** [review]

**Pull request merged**